scripts/pre-commit.sh

```diff
@@ -82,3 +82,50 @@ if [ $SHOULD_FAIL_PRECOMMIT -ne 0 ]; then
 else
   verify_no_included_apikeys
 fi
+
+# CircleCI doesn't let us split a config file into multiple, so we have a pre-commit hook that does it for us
+# so that the file doesn't become unmanageable. 
+concatenate_circleci_configs() {
```

Comment:
this is the only file with real changes aside from the original config.yml changes, so those are the only two ones worth really reviewing